earthpy/tests/test_plotrgb.py

```diff
@@ -0,0 +1,101 @@
+"""Tests for plot_rgb function"""
+
+import numpy as np
+import numpy.ma as ma
+import matplotlib.pyplot as plt
+import pytest
+import rasterio as rio
+from earthpy.plot import plot_rgb
+from earthpy.io import path_to_example
+
+plt.show = lambda: None
+
+
+@pytest.fixture
+def rgb_image():
+    """Fixture holding an RGB image for plotting"""
+    with rio.open(path_to_example("rmnp-rgb.tif")) as src:
+        rgb = src.read()
+    return rgb
+
+
+def test_1band(rgb_image):
+    """Test to ensure the input image has 3 bands to support rgb plotting"""
+    rgb_1band = rgb_image[1]
+
+    with pytest.raises(Exception):
+        plot_rgb(rgb_1band)
+
+
+def test_ax_provided(rgb_image):
+    """Test to ensure the plot works when an explicit axis is provided"""
+    f, ax1 = plt.subplots()
+    f, ax = plot_rgb(rgb_image, ax=ax1)
+
+    rgb_im_shape = rgb_image.transpose([1, 2, 0]).shape
+    the_plot_im_shape = ax.get_images()[0].get_array().shape
+    assert rgb_im_shape == the_plot_im_shape
+    plt.close(f)
+
+
+def test_ax_not_provided(rgb_image):
+    """Test to ensure that plot_rgb produces a output image on an axis
+    when an axis object is not provided."""
+
+    f, ax = plot_rgb(rgb_image)
+    rgb_im_shape = rgb_image.transpose([1, 2, 0]).shape
+    the_plot_im_shape = ax.get_images()[0].get_array().shape
+    assert rgb_im_shape == the_plot_im_shape
+
+
+def test_stretch_image(rgb_image):
+    """Test to ensure that running stretch actually stretches the data
+    to a max value of 255."""
+
+    im = rgb_image
+    np.place(im, im > 150, [0])
+
+    f, ax = plot_rgb(im, stretch=True)
+    max_val = ax.get_images()[0].get_array().max()
+    assert max_val == 255
+    plt.close(f)
+
+
+def test_stretch_image_clip(rgb_image):
+    """Test that if you run clip with a different stretch than the default,
+     you get a different mean value given the change in the range of values
+     stretched."""
+
+    im = rgb_image
+    np.place(im, im > 150, [0])
+
+    f, ax = plot_rgb(im, stretch=True, str_clip=1)
+    mean_val = ax.get_images()[0].get_array().mean()
+    assert mean_val == 110.23530766608626
```

Comment:
hey @mbjoseph I think i generally get what you did above however this doesn't test that the actual stretch is being implemented correctly. it just tests (i think mean_vals is supposed to be stretch_vals maybe?? i'm not sure above but it looks like you are just testing the number of returned values rather than the accuracy of the actual value. 

i wanted to test that the actual stretch worked. Thus i calculated what the mean should be with the val of 1 for the stretch clip. should i just include that code in the function? it will just become a bit more complex that way but i guess then you know where the number comes from. thoughts?

Reply:
ok two new tested are implemented on the new stretch helper function.